### PR TITLE
feat: edit toggle in cluster mode

### PR DESCRIPTION
**Pull request by stan-gabriel**
This PR...

## Changes
Add new Settings option: Allow editing resources

## How to test it
- 1 Go to Setting -> disable "Allow editing resources" under "Cluster Mode"
- 2 Enable "Cluster Mode"
- 3 Should not be able to edit resources

## Screenshots
image

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/molecules/Monaco/Monaco.tsx [hunk 1]
```diff
@@ -268,18 +269,18 @@ const Monaco = (props: {diffSelectedResource: () => void; applySelection: () =>
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [editor, selectedResourceId, firstCodeLoadedOnEditor]);
 
-  // read-only if we're in preview mode and another resource is selected - or if nothing is selected at all
+  // read-only if we're in preview mode and another resource is selected - or if nothing is selected at all - or allowEditInClusterMode is false
   const isReadOnlyMode = useMemo(() => {
+    if (!settings.allowEditInClusterMode && previewType === 'cluster') {
```
Comment (devcatalin) [inline src/components/molecules/Monaco/Monaco.tsx:274]:
```suggestion
    if (isInPreviewMode && previewType === 'cluster' && !settings.allowEditInClusterMode) {
```

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/components/molecules/Monaco/Monaco.tsx (hunk 1 above)
Comment (stan-gabriel) [reply src/components/molecules/Monaco/Monaco.tsx]: fixed

**Comment by olensmar** [comment]
nice! would be great if we could fix #898 together with this - to be consistent with this added option

**Comment by stan-gabriel** [comment]
> nice! would be great if we could fix #898 together with this - to be consistent with this added option
I will take a look and fix #898 as well

**Comment by devcatalin** [comment]
Merging this now as I'd prefer having #898 in a separate PR

**Pull request merged**